pandevice/base.py

```diff
@@ -1500,6 +1500,59 @@ def _cast_version_value(self, value):
         return ans
 
 
+class ParentAwareXpath(object):
+    """Class to handle xpaths of objects.
+
+    Some objects have a different xpath based on where in the tree they are
+    located.  This class allows you configure various xpaths that can vary
+    both on version and what the parent class is.
+
+    If no explicit parent is specified, then the global parent of `None' is
+    assumed.
+
+    """
+    def __init__(self):
+        self.settings = {}
+
+    def add_profile(self, version=None, value=None, *parents):
```

Comment:
This argument signature seems rigid.  It means if there is a parent, then you can't write `values='xyz'` because a keyword arg can't come before a `*arg`.  What do you think of using a signature like this instead?

    def add_profile(self, version=None, value=None, parents=())

Then expect `parents` to contain an iterable.